### ISSUE TITLE
Going to account page shows error page, always

### ISSUE THREAD

**Issue by ssg** [forem/forem#11936]
**Describe the bug**
Error page on account settings page.

**To Reproduce**
1. Login to dev.to and go to <https://dev.to/settings/account>
2. Page shows error and asks me to file a bug report

**Expected behavior**
Account page should be shown.

**Screenshots**
Standard error page with turning robot animation. 

**Desktop (please complete the following information):**
Windows 10 20H2, Edge browser

**Additional context**
My username on Dev.To is "esesci". I have login setup through GitHub integration and I feel like it might be related.

This is consistently reproducible.

**Comment by LE-HU** [comment]
Hello, is it possible to test the issue on different browsers? (disabling add-on's is a good idea)
You can also try to clear browser cache on edge.
Do you have twitter account connected to dev.to? Settings page renders that too.

**Comment by Zhao-Andy** [comment]
Hey @ssg, thanks for the report! Looks like this is an error that the settings page does not properly handle which leads to the 500.

In the settings page we try to show you your emails associated with whatever social provider you've linked (Twitter / GitHub). In this case, your Twitter account for whatever reason does not have the data needed for us to pull, and our chained method of `auth_data_dump.info.email` errors out with `undefined method info on nil 'auth_data_dump'`.

For your case specifically, I've removed your Twitter association from your account. Feel free to add it back in if you'd like via https://dev.to/settings/account.

As for the issue, here is the backtrace of the error: https://app.honeybadger.io/fault/66984/26ac848fae65a336e73fe6ea2b6d55cb

I think here we can update the view `app/views/users/_account_providers_emails.html.erb:15` to:
```
<span><%= identity&.email ? "No email found! Please try relinking your #{identity.provider} account to avoid errors." : identity.email %></span>
```

**Comment by ssg** [comment]
Thank you @Zhao-Andy for looking at this! I'm leaving this open so it can be addressed in code for other users too.

**Comment by LE-HU** [comment]
Hello, I've spent some time investigating in order to find out more about this one.
- thank you @Zhao-Andy for the logs.
- link to [Identity.email function](https://github.com/forem/forem/blob/361d09dd42c4bf510c5dd33f350ba34b068861c1/app/models/identity.rb#L41)
- I was able to replicate the issue locally by connecting a twitter account to local forem, and setting the email function to
```ruby
def email
  nil.info.email
end
```
- I can browse the whole app with the exception of /settings/account. Existing rspec identity_spec.rb catches the change.
_1) Identity#email returns the email associated with the identity
     Failure/Error: nil.info.email_

- What got my attention was the error I've got while signing up using twitter account without a image attached (really old one that I've never uploaded a profile image to). From my common sense I believe there should be a placeholder image url present in the payload in such case? I think this could be a separate issue, I'll try to replicate it.
![image](https://user-images.githubusercontent.com/19828138/102680748-6985d380-41bb-11eb-95bd-282017afd454.png)
Link to [schema](https://github.com/omniauth/omniauth/wiki/Auth-Hash-Schema).
I'd like to ask if there are other places in codebase where the functions rely on payload.info content that might be corrupted. I'm not yet familiar with the codebase, please correct me if im wrong in any part.
- I'm not sure if we'd like to implement a local workaround somewhat like the below, or the team would like to look at this at some broader spectrum to check the payload not being nil in more general fashion.
```ruby
  def email
    auth_data_dump&.info&.email ? identity.email : "No email found! Please try relinking your #{identity.provider} account to avoid errors."
  end
```
Thanks!

**Comment by Zhao-Andy** [comment]
Hi @LE-HU! Sorry for the late reply. Looks like you're off to a great start.

I think your last solution you propose makes sense! I was thinking we might want to update the user's `identity.auth_data_dump` every time they sign in to have updated information, but that's a bit more involved and not necessarily part of the scope of this issue. For now, displaying the error message you proposed seems like a good solution.

Re: the profile image issue -- agreed that it shouldn't prevent sign ups. We have an issue internally (that could probably be public too, just didn't get around to making it public) and will be fixing it soon. :)

**Comment by LE-HU** [comment]
@Zhao-Andy no problem.
I'll try to fix this one over the weekend. I've edited my previous code snippet as it had some flaw and I'd like to propose a completely valid one,
Good to know profile image issue is being tackled internally.

Will keep you posted how things develop.

**Comment by LE-HU** [comment]
Hello,
Issue can be closed, resolved by #12322
Thanks!

**Comment by ssg** [comment]
Thank you @LE-HU and @Zhao-Andy for working on this, much appreciated.
